Aux Mining Base Construction Console

GunHog opened the pull request:
- Adds the base construction console (code only)
- Adds the consoles to the maps (Box and Meta) and moves things around a little.

The base construction console is an advanced camera console, similar to the one employed in Xenobiology. It is locked into the aux mining base's building site and is used as a remote RCD for easy and fast customization of the mining base.

The console comes with a high-speed, large capacity internal RCD and the ability to build four tiny fans, for sane atmos management for up to four exits.

**NEW!** The base can now build up to four plasma cutter turrets for defense, calibrated to never target humans. Note that standing between them and a target is a very poor idea! Also make sure they have plenty of clearance, as windows and walls can block them.

I am adding this because I am disappointed in the lack of use for the base. A common complaint is that there is not enough time in the round to prepare a base and drop it. This is, hopefully, one step towards addressing that concern. The console is a way of allowing people to quickly and easily reshape and customize their base (It does NOT have to be a square!) to their liking. Engineering and Science teams can more quickly focus on setting up the interior how they please.

**Second and third sprites by @chanoc1!**

:cl: Gun Hog
add: Nanotrasen Engineering has devised a construction console to assist with building the Auxiliary Mining Base, usually located near a station's Arrivals hallway. A breakthrough in bluespace technology, this console employs an advanced internal Rapid Construction Device linked to a camera-assisted holocrane for rapid, remote construction!
/:cl:

![image](https://cloud.githubusercontent.com/assets/4955510/22718792/ee0b5d7a-ed67-11e6-9efd-ba275d933bd2.png)

Commit:
Aux Mining Base Construction Console

- Adds the base construction console (code only)
- Adds the consoles to the maps and moves things around a little.

Commit:
Makes eye visible

Hopefully this actually makes the eye visible to all players when the
console is in use.

Commit:
Merge remote-tracking branch 'remotes/upstream/master' into Aux_Base_Construction_Console

PKPenguin321 commented:
like build mode but not

Commit:
Map Fix

I hope this fixes the map issue!

optimumtact commented:
nobody uses the aux base because there's really no point tbh, there is already a base on lavaland and you can't really get anything by constructing another one

FantasticFwoosh commented:
Base takes too long to build manually, and getting that double door right means blocking the entrance door to the actual building area awkwardly.

Having a 'build computer' attached to a camera in other places like the construction room (as a mini EVA, some materials to slot into the machine in the spare boxes) attached to the station camera grid so engineers can materialise in walls slowly or when there isn't a organic within 5 or so tiles would be nice.

Building your own build computer too would be the future, a bit like slime consoles.

GunHog commented:
I have never used the Fulton stuff, explain it a little to me? Do you want the computer to spawn the beacons or the balloon thingies?

FantasticFwoosh commented:
A return beacon would be nice for cargo to send supplies directly to the aux base (though it can just arrive in person using the landing craft) eh, nevermind.

A fulton beacon recieves the bag connected to it, so a miner would need to step outside with loaded bag full of ores and send it to a beacon in cargo. 

> Besides expanding or defining a camera network in the aux base would let you build outside its normal radius, for a module next door (or just a corridor to a pod nearby lazily as a living space)

Commit:
Merge remote-tracking branch 'remotes/upstream/master' into Aux_Base_Construction_Console

# Conflicts:
#	_maps/map_files/MetaStation/MetaStation.dmm
#	_maps/map_files/TgStation/tgstation.2.1.3.dmm

Commit:
Map conflict fixes.

I hate maps. Maps suck. They are so annoying and evil. All I know is
pain.

GunHog commented:
In a future PR, I would like to have a mining shuttle dock mapped to the station's aux base doc which becomes enabled once the mining shuttle beacon is activated. This will allow the mining shuttle to dock where the mining base used to live, thus resolving another concern where players returning from the base via the shuttle could only go to Cargo.

GunHog commented:
Oh, regarding the fulton beacon: I would include one except that they need to be pre-linked in order to work. I am afraid to touch the maps more than necessary, fixing them is a huge pain. 

If there are any additional console features you would like to see considered, now is the best time to let me know!

Commit:
Turrets and tweaks

- The base console may now build up to four plasma defense turrets,
calibrated to only attack hostile mobs, never humans.

- Added construction sounds for building things with the base console.

- Fixed a bug with tiny fans placement

GunHog commented:
**New in this commit:**
- The base console may now build up to four plasma defense turrets,
calibrated to only attack hostile mobs, never humans.

- Added construction sounds for building things with the base console.

- Fixed a bug with tiny fans placement

The turrets tend to get taken out if more than one monster is attacking, but they can put up a somewhat acceptable performance.

XDTM commented:
Does the aux base do damage when landing? I could see it as a megafauna-killing strat

GunHog commented:
Go ahead. Try it. I dare you. Get a team and drop the Aux Base directly on top of a dragon. Please be sure to record it so I can laugh at you later.
## I HYPER CLOWN DARE YOU.

Commit:
Two new sprites

Credit to chanoc1

Kromgar commented:
@GunHog Move the shuttle beacon to the mining base. I didn't even know it was a thing until recently

GunHog commented:
I actually did put it inside the base now, next to the pillar.

Kromgar commented:
@GunHog Pillar? What do you mean by pillar?

GunHog commented:
The wall in the middle, I call it the "central pillar".

Commit:
Merge remote-tracking branch 'remotes/upstream/master' into Aux_Base_Construction_Console

# Conflicts:
#	icons/mob/actions.dmi

Kromgar commented:
Oh @GunHog no i didn't mean in the auxillary base. But i guess that works. Most miners don't even know the beacon exists at all. I thought it would be a nice incentive to actually put down coordinates if the beacon was at the mining base. Most miners dont even carry the coordinate beacon for the base

GunHog commented:
The base remote? It is at the top of every miner's locker. They do not carry them because miners deliberately ignore or drop the item. In my next PR, I would like to make it so that the base computer can choose a random point in Lavaland to drop if the miners never bother setting a drop point.

optimumtact commented on _maps/map_files/MetaStation/MetaStation.dmm:
```diff
@@ -1,5 +1,4 @@
 //MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE
-//Model dictionary trimmed on: 02-02-2017 02:51 (UTC)
```
you don't have to keep removing this comment

GunHog replied:
It happens automatically, I just forgot to revert it.

AnturK commented on code/modules/mining/aux_base_camera.dm:
```diff
@@ -0,0 +1,266 @@
+	fan_action.Grant(user)
+	turret_action.target = src
+	turret_action.Grant(user)
+	eyeobj.invisibility = 0 //When the eye is in use, make it visible to players so they know when someone is building.
```
If you're doing this, give it IC sprite. Construction crane or something similar.

RemieRichards replied:
little floating robot.

GunHog replied:
If I had such a sprite, I would happily do so!

GunHog replied:
I have an idea that might work splendidly! I will get on it soon!

chanoc1 replied:
someone say sprites?

GunHog replied:
I have an excellent unused sprite from @WJohn I am going to try out!

chanoc1 replied:
cool

AnturK reviewed (approved):
Looks good, just some minor suggestion

Commit:
Construction Drone Sprite!

- Adds the construction drone sprite to replace the visible [ ]
indicator!
- Small tweaks and fixes to it!
Sprites by WJohnston!

GunHog commented:
@RemieRichards, may I present to you the little robot concept:
![screenshot 2017-02-10 16 18 47](https://cloud.githubusercontent.com/assets/4955510/22846808/fae9391a-efae-11e6-8493-88f35f83624a.png)

This little sprite drone buddy is now what players will see when someone is using the building console! It even has movement sprites and animation! I extracted them from the mining borg PR where @WJohn made me these beautiful drone sprites that sadly went unmerged :cry:. Now, they have a new purpose and outlook on life!

coiax commented:
I HATE TINY FANS SO MUCH

But this is a good idea.

Commit:
Turret flavor

- Added fluff text to the turrets!
